src/services/dog.js

```diff
@@ -0,0 +1,39 @@
+import { checkError, client } from './client';
+
+export async function fetchDogs() {
+  const resp = await fetch(`${process.env.REACT_APP_SUPABASE_URL}/rest/v1/dogs?`, {
+    headers: {
+      apikey: process.env.REACT_APP_SUPABASE_KEY,
+      Authorization: `Bearer ${process.env.REACT_APP_SUPABASE_KEY}`,
+    },
+  });
+  const data = await resp.json();
+  return data;
+}
+
+export async function fetchDogById(id) {
+  const resp = await client.from('dogs').select('*').match({ id }).single();
+  return checkError(resp);
+}
+
+export async function updateDog(id, name, image, age, breed, bio) {
+  const resp = await client.from('dogs').update({ name, image, age, breed, bio }).eq('id', id);
+  return checkError(resp);
+}
+
+export async function createDog(id, name, image, age, breed, bio) {
```

Comment:
We shouldn't need an id to insert into our db. The id should be automatically generated by through supabase